sdk/formrecognizer/azure-ai-formrecognizer/azure/ai/formrecognizer/_form_recognizer_client.py

```diff
@@ -0,0 +1,191 @@
+# coding=utf-8
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+
+# pylint: disable=protected-access
+
+from typing import (  # pylint: disable=unused-import
+    Any,
+    IO,
+    TYPE_CHECKING,
+)
+import six
+from azure.core.tracing.decorator import distributed_trace
+from azure.core.polling import LROPoller
+from azure.core.polling.base_polling import LROBasePolling  # pylint: disable=no-name-in-module,import-error
+from azure.core.pipeline.policies import AzureKeyCredentialPolicy
+from ._generated._form_recognizer_client import FormRecognizerClient as FormRecognizer
+from ._response_handlers import (
+    prepare_receipt_result,
+    prepare_layout_result,
+)
+from ._generated.models import AnalyzeOperationResult
+from ._helpers import get_content_type, POLLING_INTERVAL, COGNITIVE_KEY_HEADER
+if TYPE_CHECKING:
+    from azure.core.credentials import AzureKeyCredential
+
+
+class FormRecognizerClient(object):
+    """FormRecognizerClient.
+
+    :param str endpoint: Supported Cognitive Services endpoints (protocol and hostname,
+        for example: https://westus2.api.cognitive.microsoft.com).
+    :param credential: Credentials needed for the client to connect to Azure.
+        This is an instance of AzureKeyCredential if using an API key.
+    :type credential: ~azure.core.credentials.AzureKeyCredential
+    """
+
+    def __init__(self, endpoint, credential, **kwargs):
```

Comment:
More a question, but why does FR not allow TokenCredentials

Reply:
They do support AAD auth, but there's a bug in the platform right now that won't get fixed until after preview 1. So we will not expose this type of auth yet.